clang/test/CodeGenSYCL/sycl-post-link-used-attr.cpp

```diff
@@ -0,0 +1,31 @@
+// RUN: %clang_cc1 -disable-llvm-passes %s -emit-llvm -o - | sycl-post-link -split=auto -ir-output-only -o %t.out_ir_only.bc
```

Comment:
@AidanBeltonS, `sycl-post-link` tool is an LLVM tool, so the test should be llvm/test/tools/sycl-post-link/ and input file should be LLVM IR. It should be easy to create from the clang output. Please, take a look at other tests 
from llvm/test/tools/sycl-post-link/.

Reply:
Using `sed` is also not a good idea, I think, because it might not be available in all environments we test (like Windows, for example)

Reply:
Okay, that makes sense. There is an already existing test `erase_used.ll` in llvm/tools/sycl-post-link/ which performs a similar task. Is there a preference for adding a new test or extending an existing one?

Reply:
Extending `erase_used.ll` sounds like a good option to me.